pkg/Cpanel/Security/Advisor/Assessors/Trojans.pm

```diff
@@ -73,7 +91,618 @@ sub _check_for_libkeyutils {
     );
 
     return 1;
+}
+
+sub _check_for_UMBREON_rootkit {
+    my ($self) = @_;
+    my $dir = '/usr/local/__UMBREON__';
+    my $dir2 = '/usr/local/UMBREON';
+    if ( chdir $dir or chdir $dir2 ) {
+        $self->add_bad_advice(
+            'text'       => [ "UMBREON rootkit check: Evidence of the UMBREON rootkit was found." ],
+            'suggestion' => [
+                'Check the following to determine if this server is compromised "[output,url,_1,Determine your Systems Status,_2,_3]"',
+                'https://documentation.cpanel.net/display/CKB/Determine+Your+System%27s+Status',
+                'target',
+                '_blank'
+            ],
+        );
+    }
+}
+
+sub _check_for_jynx2_rootkit {
+    my ($self) = @_;
+    my $dir = '/usr/bin64';
+    my @found_jynx2_files = ();
+    my @jynx2_files = qw( 3.so 4.so );
+    for (@jynx2_files) { 
+        my $file = $dir . "/" . $_;
+        if ( -e $file ) { 
+            push( @found_jynx2_files, $file );
+        }
+    }
+    if ( scalar @found_jynx2_files > 0 ) {
+        $self->add_bad_advice(
+            'text'       => [ "Jynx 2 rootkit check: Evidence of the Jynx 2 rootkit was found." ],
+            'suggestion' => [
+                'Check the following to determine if this server is compromised "[output,url,_1,Determine your Systems Status,_2,_3]"',
+                'https://documentation.cpanel.net/display/CKB/Determine+Your+System%27s+Status',
+                'target',
+                '_blank'
+            ],
+        );
+    }
+}
+
+sub _check_for_cdorked_A_rootkit {
+    my ($self) = @_;
+    my $apache_bin   = '/usr/local/apache/bin/httpd';
+    my $max_bin_size = 10_485_760;                      # avoid slurping too much mem
+    return if ( !-f $apache_bin );
+    return if ( ( stat($apache_bin) )[7] > $max_bin_size );
+
+    my $has_cdorked = 0;
+    my $signature;
+    my @apache_bins = ();
+    push @apache_bins, $apache_bin;
+    my @PROCESS_LIST;
+    for my $process (@PROCESS_LIST) {
+        if ( $process =~ m{ \A root \s+ (\d+) [^\d]+ $apache_bin }xms ) {
```

Comment:
I recommend adding `/a` to the regex to keep `\d` from matching non-ASCII numbers; see "Character set modifiers" at http://perldoc.perl.org/perlre.html.  This also improves performance, as the number of characters to check for is reduced.  For a list of numeric characters in Unicode, see http://www.fileformat.info/info/unicode/category/Nd/list.htm.

```perl
if ( $process =~ m{ \A root \s+ (\d+) [^\d]+ $apache_bin }xmsa ) {
```